### ISSUE TITLE
Unordered lists in Firefox

### ISSUE THREAD

**Issue by ryanfeeley** [jhildenbiddle/vertical-rhythm-reset#23]
I wondered why the grid was off and scrolled up to here to find the culprit. Seems to only affect Firefox.
<img width="330" alt="image" src="https://user-images.githubusercontent.com/68213/148591954-92dde4b7-80fa-4e74-84e8-3f0ff3cb2d9c.png">

**Comment by jhildenbiddle** [comment]
Hi @ryanfeeley.

TL;DR: Can't be fixed AFAIK, but options exist to work around the issue.

From what I can tell, the issue is cause by Firefox's insistence on rendering list item bullets using the `-moz-bullet-font` instead of inheriting the font specified for the list item. Notice in the following screenshot that `::marker` pseudo-element within the ordered list inherits the `font-family` value (`sans-serif`) of its parent `<li>` element:

![19-ol](https://user-images.githubusercontent.com/442527/149836016-4250b8bf-d18a-4c58-b65a-50340d4b6743.jpg)

Now, notice the `font-family` of the `::marker` element within the unordered list:

![19-ul](https://user-images.githubusercontent.com/442527/149836375-b489a932-a7cb-4e06-af02-fc97650d40f6.jpg)

Why is Firefox setting `font-family` to `-moz-bullet-font` instead of `sans-serif`? Who knows. The good news is that we can "fix" the spacing issue by forcing the `::marker` element to inherit the `font-family` value:

```css
ul ::-moz-list-bullet {
  font-family: inherit;
}
```

Unfortunately, this change causes Firefox to render the bullets at the wrong size and position:

**Before:**
<img width="340" alt="CleanShot 2022-01-17 at 15 04 30@2x" src="https://user-images.githubusercontent.com/442527/149837306-2d7004ea-86b4-434b-8beb-8f48c65df946.png">

**After:**
<img width="340" alt="CleanShot 2022-01-17 at 15 05 07@2x" src="https://user-images.githubusercontent.com/442527/149837317-cecb6664-8dd3-4513-a090-0dcb7f5f8c58.png">

The `font-size` can be adjusted, but doing so reintroduces the height issue and new vertical alignment issues:

```css
ul ::-moz-list-bullet {
  font-family: inherit;
  font-size: 1.5em;
}
```
<img width="340" alt="CleanShot 2022-01-17 at 15 18 38@2x" src="https://user-images.githubusercontent.com/442527/149838018-5c555550-5a10-45a9-b458-de3c99cd92a8.png">

No amount of tweaking to properties like `line-height` or `vertical-align` help, so "fixing" the `font-family` value isn't really a great solution unless you're okay with the relatively small bullets in Firefox.

Another possible solution is to manually render a bullet for `::-moz-list-bullet` pseudo-elements using the `content` property:

```css
ul ::-moz-list-bullet {
  content: '●  ';
}
```
<img width="316" alt="CleanShot 2022-01-17 at 15 33 12@2x" src="https://user-images.githubusercontent.com/442527/149839197-29c58d8e-7a4e-43ee-8e52-fb83ed9ba333.png">

Unfortunately, this isn't a fix I can bake into the project because it isn't easy to override without knowing what's happening under the hood and it doesn't work well with nested lists.

I think this is worth filing a bug with Mozilla. I will add more to this thread if/when I know more. 

Happy to answer any additional questions you may have. Until then, I will close the issue since there is nothing for me to do.

Thanks!

**Comment by jhildenbiddle** [comment]
Firefox bug filed here: https://bugzilla.mozilla.org/show_bug.cgi?id=1750905
